### PR TITLE
Sanitize imports so tools.datetimes, tools.timedeltas, tools.numeric

### PR DESCRIPTION
**Pull request by jbrockmendel**
are robust to changes in import order.

- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/tools/numeric.py [hunk 1]
```diff
@@ -158,12 +158,13 @@ def to_numeric(arg, errors='raise', downcast=None):
                     if values.dtype == dtype:
                         break
 
+    from pandas import Series, Index
```
Comment (gfyoung) [inline pandas/core/tools/numeric.py]: Let's just import `Series` and `Index` when we need to (i.e. import `Series` under `is_series` and `Index` under `is_index`.

Comment (jbrockmendel) [inline pandas/core/tools/numeric.py]: Just did that... and then went and expanded the scope a whole bunch.

**Comment by jorisvandenbossche** [comment]
Is this actually the way we want to go? Eg the plotting modules should be a rather independent module from core, so what's the problem with importing from there? 
Replacing all occurrences of Index to ABCIndexClass and similar, IMO makes the code less readable and more difficult for new contributors.

**Comment by jbrockmendel** [comment]
> so what's the problem with importing from there?

Until recently there were issues with circular imports from `plotting`.  This started out as an attempt to preemptively untangle any potential similar issues, but lost focus and got away from me.  You're right this is not a good use of time.  My bad.

**Comment by jreback** [comment]
> Is this actually the way we want to go? Eg the plotting modules should be a rather independent module from core, so what's the problem with importing from there?
Replacing all occurrences of Index to ABCIndexClass and similar, IMO makes the code less readable and more difficult for new contributors.

yes. this is must easier to handle imports, and only when actual construction is need do you actually import Series/DataFrame. 

this already exists in most of the codebase.

**Pull request closed without merging**